### PR TITLE
test(browser) : enhance test coverage for http and auth params utilities

### PR DESCRIPTION
**Pull request by KavishkaVenuka**
- Add comprehensive test cases for hasAuthParamsInUrl utility
- Add extensive test coverage for http utility including:
  - Request method testing with various HTTP methods
  - RequestAll method testing for batch operations
  - Edge case handling
  - Error scenarios
  - Integration tests
  - Concurrent request handling

This change improves the test coverage and reliability of the browser package's utilities.

### Purpose
-This PR enhances the test coverage for utilities in the browser package by adding comprehensive test suites for both hasAuthParamsInUrl and HTTP utilities. The changes focus on improving code reliability and ensuring robust error handling.

###Key Improvements:
-Enhanced test coverage for hasAuthParamsInUrl utility

###URL parameter validation
-Edge case handling
-Special character testing
-Parameter positioning scenarios

###Comprehensive HTTP utility test coverage
-Various HTTP method validations (GET, POST, PUT, DELETE)
-Batch operation testing via RequestAll method
-Concurrent request handling scenarios
-Error case validation
-Integration test scenarios
-Edge case handling for various request states

###Related Issues
-N/A

###Related PRs
-N/A

###Checklist
 -Followed the [CONTRIBUTING] guidelines
 -Manual test round performed and verified
 -Unit tests provided in:
---->[hasAuthParamsInURL.test.ts]
---->packages/browser/src/utils/__tests__/http.test.ts
 -Documentation provided (No documentation updates required as this is test-only changes)

###Security checks
 -Followed secure coding standards in [http://wso2.com/technical-reports/wso2-secure-engineering-guidelines]
 -Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

###Additional Notes
-All test cases follow the project's testing conventions
-Tests are designed to be maintainable and readable
-No production code changes were made, only test improvements

### CODE REVIEW

File: packages/browser/jest.config.js
Comment (brionmario) [inline packages/browser/jest.config.js]: Hey @KavishkaVenuka,

Could you please explain why we need a Jest Config?

File: packages/browser/package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "playwright": "^1.52.0",
     "prettier": "^2.6.2",
     "rimraf": "^6.0.1",
-    "typescript": "~5.7.2",
+    "typescript": "~5.7.3",
```
Comment (brionmario) [inline packages/browser/package.json]: Any reason for this version bump?

File: packages/browser/jest.config.js
Comment (KavishkaVenuka) [reply packages/browser/jest.config.js]: My bad sir. I'll remove it.

File: packages/browser/package.json [hunk 2]
```diff
@@ -74,4 +74,4 @@
   "publishConfig": {
     "access": "public"
   }
-}
+}
```
[the remark this answers is not in the document]
Comment (KavishkaVenuka) [reply packages/browser/package.json]: Done sir.

**Comment by brionmario** [comment]
Closing this PR as part of the stale PR cleanup.
